tests/test_context.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+import pytest
+from plumbum import local
+
+import copier
+
+from .helpers import build_file_tree, git_save
+
+
+def test_exclude_templating_with_operation(tmp_path_factory: pytest.TempPathFactory) -> None:
+    """
+    Ensure it's possible to create one-off boilerplate files that are not
+    managed during updates via _exclude using the ``_operation`` context variable.
+    """
+    src, dst, dst2 = map(tmp_path_factory.mktemp, ("src", "dst", "dst2"))
+
+    template = r"{%- if _operation == 'update' %}boilerplate{%- endif %}"
+    with local.cwd(src):
+        build_file_tree(
+            {
+                "copier.yml": f"_exclude:\n - \"{template}\"",
+                "{{ _copier_conf.answers_file }}.jinja": "{{ _copier_answers|to_yaml }}",
+                "boilerplate": "foo",
+                "other_file": "foo",
+            }
+        )
+        git_save(tag="1.0.0")
+        build_file_tree(
+            {
+                "boilerplate": "bar",
+                "other_file": "bar",
+            }
+        )
+        git_save(tag="2.0.0")
+    boilerplate = dst / "boilerplate"
+    other_file = dst / "other_file"
+
+    copier.run_copy(str(src), dst, defaults=True, overwrite=True, vcs_ref="1.0.0")
+    for file in (boilerplate, other_file):
+        assert file.exists()
+        assert file.read_text() == "foo"
+
+    with local.cwd(dst):
+        git_save()
+
+    copier.run_update(str(dst), overwrite=True)
+    assert boilerplate.read_text() == "foo"  # This file is excluded from updates
+    assert other_file.read_text() == "bar"
+
+    # After using the worker for an `update` operation, reuse it for a `copy` again.
+    # This checks that the cached `match_exclude` property is regenerated
+    # after a context switch back from update to copy.
+    copier.run_copy(str(src), dst2, defaults=True, overwrite=True, vcs_ref="1.0.0")
+    for filename in ("boilerplate", "other_file"):
+        assert (dst2 / filename).exists()
+        assert (dst2 / filename).read_text() == "foo"
```

Comment:
Yeah, it's a completely theoretical issue at the moment as described [here](https://github.com/copier-org/copier/pull/1733#issuecomment-2463617397). This was intended to warn future developers in case this is changed at some point. Should I remove it?